ModelBundle/Validator/Constraints/CheckMainAliasPresenceValidator.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace OpenOrchestra\ModelBundle\Validator\Constraints;
+
+use OpenOrchestra\ModelInterface\Model\SiteInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+class CheckMainAliasPresenceValidator extends ConstraintValidator
+{
+    /**
+     * Checks if the passed value is valid.
+     *
+     * @param SiteInterface $value The value that should be validated
+     * @param Constraint    $constraint The constraint for the validation
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$value->getMainAlias()->isMain()) {
```

Comment:
I agree with the idea of throwing an exception, but I don't think it introduces a BC Break, it only introduce one more control